tests/unit/splash/test_splash_model.py

```diff
@@ -43,19 +43,28 @@ def test_splash_model_init(splash_core_constants, grid_benchmarks, var, flag):
     )
 
     ds = grid_benchmarks[0].sel(time=slice("2000-01-01", "2000-04-01")).copy()
+    dates = ds.time.data
 
-    vmin, vmax = bounds[var]
-    arr = ds[var].data
-    if flag == "underflow":
-        arr.flat[np.random.choice(arr.size)] = vmin - 1e-4
-    elif flag == "overflow":
-        arr.flat[np.random.choice(arr.size)] = vmax + 1e-4
+    if var == "dates":
+        if flag < 0:
+            dates = dates[:-1]
+        else:
+            ds = ds.sel(time=dates[:-1])
+        context = pytest.raises(ValueError)
+    else:
+        vmin, vmax = bounds[var]
+        arr = ds[var].data
+        if flag < 0:
+            arr.flat[np.random.choice(arr.size)] = vmin - 1e-4
+        else:
+            arr.flat[np.random.choice(arr.size)] = vmax + 1e-4
+        context = pytest.warns(UserWarning)
```

Comment:
I'd generally go for more test functions (even if repetitious) rather than create branching patterns within tests, but this isn't the most complex internals I've ever seen 😄. If you are going to do this, it does make sense to add brief comments to explain what each branch does.

Reply:
I've added the comments now.